packages/block-library/src/utils/interactivity/vdom.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * External dependencies
+ */
+import { h } from 'preact';
+/**
+ * Internal dependencies
+ */
+import { directivePrefix as p } from './constants';
+
+const ignoreAttr = `${ p }ignore`;
+const islandAttr = `${ p }island`;
+const directiveParser = new RegExp( `${ p }([^.]+)\.?(.*)$` );
+
+export const hydratedIslands = new WeakSet();
+
+// Recursive function that transforms a DOM tree into vDOM.
+export function toVdom( root ) {
+	const treeWalker = document.createTreeWalker(
+		root,
+		205 // ELEMENT + TEXT + COMMENT + CDATA_SECTION + PROCESSING_INSTRUCTION
+	);
+
+	function walk( node ) {
+		const { attributes, nodeType } = node;
+
+		if ( nodeType === 3 ) return [ node.data ];
+		if ( nodeType === 4 ) {
+			const next = treeWalker.nextSibling();
+			node.replaceWith( new window.Text( node.nodeValue ) );
+			return [ node.nodeValue, next ];
+		}
+		if ( nodeType === 8 || nodeType === 7 ) {
+			const next = treeWalker.nextSibling();
+			node.remove();
+			return [ null, next ];
+		}
```

Comment:
the node constants I believe are going to be present _everywhere_ we have a DOM, so why not use them here? I cannot for the life of me remember the difference between `4` and `7`

```js
switch ( nodeType ) {
	case Node.TEXT_NODE:
		return [ node.data ];

	case Node.CDATA_SECTION_NODE: {
		const next = treeWalker.nextSibling();
		node.replaceWith( new window.Text( node.nodeValue ) );
		return [ node.nodeValue, next ];
	}

	case Node.PROCESSING_INSTRUCTION_NODE:
	case Node.COMMENT_NODE: {
		const next = treeWalker.nextSibling();
		node.remove();
		return [ null, next ];	
	}
}
```

the code in here makes more sense to me seeing these

Reply:
I agree 😅 I think @luisherranz decided to do so because it saves some bytes to the client, as those props are not mangled.

Reply:
We can add them as comments.

Reply:
hm. Are we concerned that much about a few tens of bytes when we're skipping KBs?

> as those props are not mangled

that's the point 🙃
comments at a minimum would be nice. I think we'd be hard pressed to make a case that it matters that we not use the constants that are there 🤷‍♂️

Reply:
🤔 🤷‍♂️